applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1144,10 +1144,17 @@ public function calculate(&$discussion) {
             $discussion->Read = false;
             $discussion->CountUnreadComments = true;
         } else {
+            // If the category was marked explicitly read at some point, see if that applies here
             if ($category && !is_null($category['DateMarkedRead'])) {
-                // If the category was marked explicitly read at some point, see if that applies here
-                $discussion->DateLastViewed = self::maxDate($discussion->DateLastViewed, $category['DateMarkedRead']);
+                // If the discussion was created after the category was marked read and hasn't been viewed, reset DateLastViewed to null
+                if ($category['DateMarkedRead'] <= $discussion->DateInserted && is_null($discussion->CountCommentWatch)) {
```

Comment:
At this point, I'm pretty sure `DateMarkedRead` and `DateInserted` are strings (or possibly `null` for the latter). We're comparing two strings, which will usually work, but not for the reasons we're expecting and can potentially be unpredictable. PHP doesn't have any special handling for date strings, specifically. We could translate the string date value into a numerical timestamp value, using [`strtotime`](https://www.php.net/strtotime), which should give us more confidence when comparing the two. We're generating a standard numerical representation of a date, which can then be compared against other standard numerical representations.

Reply:
If `$discussion->CountCommentWatch` is `null`, then wouldn't `$discussion->DateLastViewed` also be `null` already?